Board flat in game_mechanics not just choose_move

neuromantic99 opened the pull request:
Also added a load more tests to make sure e.g. reward_function is not broken

neuromantic99 committed:
Board flat in game_mechanics not just choose_move

Also added a load more tests to make sure e.g. reward_function is not
broken

Henry-Pulver commented on delta_wild_tictactoe/game_mechanics.py:
```diff
@@ -16,16 +15,9 @@
-    return int(is_winner(board))
+def reward_function(board: List[str]) -> int:
+    winner = is_winner(board)
+    return int(winner)
```
```suggestion
    return int(is_winner(board))
```

neuromantic99 committed:
docstrings

neuromantic99 committed:
updated readme

neuromantic99 committed:
added hgm

Henry-Pulver commented on delta_wild_tictactoe/game_mechanics.py:
```diff
@@ -34,8 +26,17 @@ def choose_move_randomly(board: List[str]) -> Tuple[int, str]:
-def mark_square(board: List[List[str]], row: int, col: int, counter: str) -> List[List[str]]:
-    board[row][col] = counter
+def mark_square(
+    board: List[str], position: int, counter: str, copy_board: bool = False
```
Personally, I would just always `.copy()` this as I don't see this game ever being _too slow_

Henry-Pulver commented on tests/test_fuctions.py:
```diff
@@ -0,0 +1,103 @@
+
+def test_mark_square_do_not_copy():
+    board = get_empty_board()
+    for _ in range(100):
+        position = random.randint(0, 8)
+        counter = random.choice([Cell.X, Cell.O, Cell.EMPTY])
+        new_board = mark_square(board, position, counter, copy_board=False)
```
Maybe I'm wrong, but it seems an odd choice to allow a 'game' that continues for 100 steps because `mark_square()` doesn't check that the space is empty before placing there. Imo could lead to bugs/confusion in user code

Henry-Pulver reviewed (commented):
Love, love, LOVE the addition of tests :heart_eyes: 

My suggestions are small tweaks only

neuromantic99 committed:
Addressed Henry Comments

mark_square -> place_counter. Which will only place valid moves, rather
than just marking the board with anything

neuromantic99 committed:
remove print
